packages/flutter/lib/src/rendering/sliver_group.dart

```diff
@@ -213,15 +214,31 @@ class RenderSliverMainAxisGroup extends RenderSliver with ContainerRenderObjectM
         double childScrollOffset = 0.0;
         RenderSliver? current = childBefore(child as RenderSliver);
         while (current != null) {
-          childScrollOffset += current.geometry!.scrollExtent;
+          // If the current sliver is not the first child, add twice the scrollExtent to the scrollOffset
+          // to account for both the current and the previous slivers.
+          // If the current sliver is the first child, add its scrollExtent once to the scrollOffset
+          // to account for only the current sliver.
+          if (childBefore(current) != null) {
+            childScrollOffset += 2 * current.geometry!.scrollExtent;
+          } else if (!(childAfter(child) != null && current.geometry!.hasVisualOverflow)) {
+            childScrollOffset += current.geometry!.scrollExtent;
+          }
           current = childBefore(current);
         }
         return childScrollOffset;
       case GrowthDirection.reverse:
         double childScrollOffset = 0.0;
         RenderSliver? current = childAfter(child as RenderSliver);
         while (current != null) {
-          childScrollOffset -= current.geometry!.scrollExtent;
+          // If the current sliver is not the last child, subtract twice the scrollExtent to the scrollOffset
```

Comment:
@Mairramer to continue from the thread below, my current thought process is to revert this block and land the PR (to avoid over-engineering). Then, if it ever happens that someone runs into an issue here, their particular use case would give more clarity on how (or even if) it should be fixed.